components/remote_settings/src/remotesettings.udl

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace remotesettings {};
+
+[Custom]
+typedef string JsonObject;
+
+dictionary RemoteSettingsConfig {
+    string? server_url;
+    string? bucket_name;
+    string collection_name;
+};
+
+[Error]
+enum RemoteSettingsError {
+    "NetworkFailure",
+    "ClientError",
```

Comment:
I *think* that all of these end up as enums with strings? If so, communicating the sub-variants might be tricky - I'm thinking specifically of `ClientError::Backoff(duration)` and whether the Kotlin caller will be able to sanely handle that and extract the duration. I don't think this needs to block though because I suspect a first version will not care, but might be worth opening an issue?

(This certainly would not be the first crate with this kind of issue though!)

Reply:
AFAICT Fenix was not doing anything with the `NimbusError::Backoff` before it was moved into `rs_client::ClientError::Backoff`, so I don't suspect that this will be required for the first version either